Miracast/MiracastService/MiracastController.cpp

```diff
@@ -0,0 +1,1968 @@
+        discover_devices();
+    }
+    else
+    {
+        MIRACASTLOG_INFO("!!! BACKEND P2P DISCOVERY HAS DISABLED !!!");
+    }
+}
+
+void MiracastController::event_handler(P2P_EVENTS eventId, void *data, size_t len )
+{
+    CONTROLLER_MSGQ_STRUCT controller_msgq_data = {0};
+    std::string event_buffer;
+    MIRACASTLOG_TRACE("Entering...");
+
+    {
+        event_buffer = (char *)data;
+        free(data);
+    }
+
+    if (nullptr != m_controller_thread){
+        controller_msgq_data.msg_type = P2P_MSG;
+        controller_msgq_data.state = convertP2PtoSessionActions(eventId);
+        strcpy(controller_msgq_data.msg_buffer, event_buffer.c_str());
+
+        MIRACASTLOG_INFO("event_handler to Controller Action[%#08X] buffer:%s  ", controller_msgq_data.state, event_buffer.c_str());
+        m_controller_thread->send_message(&controller_msgq_data, sizeof(controller_msgq_data));
+        MIRACASTLOG_VERBOSE("event received : %d buffer:%s  ", eventId, event_buffer.c_str());
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+}
+
+MiracastError MiracastController::set_WFDParameters(void)
+{
+    MIRACASTLOG_TRACE("Entering...");
+    MiracastError ret = MIRACAST_FAIL;
+    if (nullptr != m_p2p_ctrl_obj){
+        ret = m_p2p_ctrl_obj->set_WFDParameters();
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return ret;
+}
+
+MiracastError MiracastController::discover_devices(void)
+{
+    MIRACASTLOG_TRACE("Entering...");
+    MiracastError ret = MIRACAST_FAIL;
+    if (nullptr != m_p2p_ctrl_obj){
+        ret = m_p2p_ctrl_obj->discover_devices();
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return ret;
+}
+
+MiracastError MiracastController::stop_discover_devices(void)
+{
+    MIRACASTLOG_TRACE("Entering...");
+    MiracastError ret = MIRACAST_FAIL;
+    if (nullptr != m_p2p_ctrl_obj){
+        ret = m_p2p_ctrl_obj->stop_discover_devices();
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return ret;
+}
+
+MiracastError MiracastController::connect_device(std::string device_mac , std::string device_name )
+{
+    MIRACASTLOG_TRACE("Entering...");
+    MIRACASTLOG_VERBOSE("Connecting to the MAC - %s", device_mac.c_str());
+    MiracastError ret = MIRACAST_FAIL;
+
+    if (nullptr != m_p2p_ctrl_obj)
+    {
+        ret = m_p2p_ctrl_obj->connect_device(device_mac);
+        if (MIRACAST_OK == ret )
+        {
+            set_WFDSourceMACAddress(device_mac);
+            set_WFDSourceName(device_name);
+        }
+        else
+        {
+            MIRACASTLOG_ERROR("!!! P2P Connect Failure for Device[%s - %s] !!!",device_name.c_str(),device_mac.c_str());
+            if ( nullptr != m_notify_handler )
+            {
+                m_notify_handler->onMiracastServiceClientConnectionError( device_mac , device_name , MIRACAST_SERVICE_ERR_CODE_P2P_CONNECT_ERROR );
+            }
+        }
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return ret;
+}
+
+#if 0
+MiracastError MiracastController::start_streaming()
```

Comment:
do we    need this?

Reply:
It is not required. I removed it as part of https://github.com/rdkcentral/rdkservices/commit/019f3d11389a6271609f0a3153b3a3984f89ca64